package.json

```diff
@@ -4,7 +4,7 @@
   "main": "build/index.js",
   "scripts": {
     "build": "grunt --verbose",
-    "prepublish": "npm run build"
+    "prepublish": "npm run build && cp react-icons/*.js ."
```

Comment:
Hmm, are we dropping all individual icon components into the package's root dir prior to publishing? That seems a bit unfortunate, especially if we don't have a cleanup job -- they're probably going to linger afterwards, requiring the dev to remove them manually...

Can we continue to keep things contained in dedicated build directories?

Reply:
Yes, they are moved to the root of the package only before `npm publish` (so it doesn't pollute it during development). We need that to do:

    import GridiconExternal from 'gridicons/external';

If they are under build, we'll need to include the `build` dir in the npm package and the individual icons would be imported like:

    import GridiconExternal from 'gridicons/build/external';

AFAIK, it is common practice for libraries that want to export individual components -such as Lodash- to move them to the root for publishing.

One thing we could do, though, is to run a clean task in the `postpublish` npm hook.

Reply:
A `postpublish` cleanup hook is somewhat better, but still not perfect 😁-- as e.g. evidenced by [this thread](https://github.com/Automattic/gridicons/pull/283#discussion_r172834729).

Too bad `package.json`'s `main` field can't be a directory. Related: https://github.com/nodejs/node/issues/14970

Another idea: Have `build/index.js` do

```js
export { default as GridiconExternal } from './external';
export { default as GridiconFoo } from './foo';
export { default as GridiconGarb } from './garb';
```

If we publish not only the transpiled but also the untranspiled code to npm, consumers can then

```js
import { GridiconExternal, GridiconGarb } from 'gridicons';
```

For projects that use webpack >= v3, tree shaking will then ensure they're only importing the relevant code (omitting the Gridicons that aren't `import`ed).

Reply:
You raised a very good point about `cjs` and `esm` with https://github.com/Automattic/gridicons/pull/284 My response to that is a9010c937f93b56596c99721b2a78f5fefa967db :)

I still prefer to publish the icons separately, as not all consumers are using a module bundler that supports tree-shaking at the moment, and I'd want to impose that on them.